src/main/groovy/de/dkfz/roddy/execution/io/LocalExecutionHelper.groovy

```diff
@@ -117,23 +146,51 @@ class LocalExecutionHelper {
      */
     static ExecutionResult executeCommandWithExtendedResult(
             String command,
+            Boolean waitFor,
+            Duration timeout = Duration.ZERO,
             OutputStream outputStream = null,
             ExecutorService executorService = executorService) {
         List<String> bashCommand = ["bash", "-c", command]
-        logger.postRareInfo("Executing the command ${command} locally.")
+        logger.postRareInfo("Executing the command ${bashCommand} locally.")
         ProcessBuilder processBuilder = new ProcessBuilder(bashCommand)
         Process process = processBuilder.start()
-        Future<List<String>> stdout
+        Future<List<String>> stdoutF
         if (outputStream == null)
-            stdout = asyncReadStringStream(process.inputStream, executorService)
+            stdoutF = asyncReadStringStream(process.inputStream, executorService)
         else
-            stdout = asyncReadStringStream(process.inputStream, executorService, outputStream)
-        Future<List<String>> stderr = asyncReadStringStream(process.errorStream, executorService)
-        Future<Integer> exitCode = CompletableFuture.supplyAsync({
-            process.waitFor()
+            stdoutF = asyncReadStringStream(process.inputStream, executorService, outputStream)
+        Future<List<String>> stderrF =
+                asyncReadStringStream(process.errorStream, executorService)
+        Future<Integer> exitCodeF = CompletableFuture.supplyAsync({
```

Comment:
Using the correct generic would eliminate the need for casting at the end and makes it more readable. But I think that this type is not really necessary here, so the cast could just be dropped.

Reply:
Correct. The `as Supplier` is unnecessary. Removed. I also changed `Future` to `CompletableFuture`.